src/wallet/scriptpubkeyman.h

```diff
@@ -246,6 +246,8 @@ class LegacyScriptPubKeyMan : public ScriptPubKeyMan, public FillableSigningProv
     std::set<int64_t> set_pre_split_keypool GUARDED_BY(cs_wallet);
     int64_t m_max_keypool_index GUARDED_BY(cs_wallet) = 0;
     std::map<CKeyID, int64_t> m_pool_key_to_index;
+    // Tracks keypool indexes to CKeyIDs of keys that have been taken out of the keypool but may be returned to it
+    std::map<int64_t, CKeyID> m_reserved_key_to_index;
```

Comment:
In commit "Key pool: Change ReturnDestination interface to take address instead of key" (06d411488c1c669f00a1c947416267941dbb7f7b)

Should there be separate map for internal / external key indexes? Or should map key be an (index_number, is_internal) pair? Otherwise how do you know the internal/external index numbers don't overlap?

Reply:
The indexes do not overlap. See https://github.com/bitcoin/bitcoin/blob/master/src/wallet/scriptpubkeyman.cpp#L1083

Reply:
as noted a few days ago, this should be renamed

Reply:
Renamed it.